lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1828,27 +1830,12 @@ estimateTxSize era skeleton =
 
     -- Total number of signatures the scripts require
     numberOf_MintingWitnesses
-        = sumVia scriptRequiredKeySigs txMintOrBurnScripts
+        = fromIntegral $
+        sumVia estimateMinWitnessRequiredPerInput txMintOrBurnScripts
 
     numberOf_ScriptVkeyWitnesses
-        = maybe 0 scriptRequiredKeySigs txPaymentTemplate
-
-    scriptRequiredKeySigs :: Num num => Script object -> num
-    scriptRequiredKeySigs = \case
-        RequireSignatureOf _ ->
-            1
-        RequireAllOf ss ->
-            sumVia scriptRequiredKeySigs ss
-        RequireAnyOf ss ->
-            sumVia scriptRequiredKeySigs ss
-        ActiveFromSlot _ ->
-            0
-        ActiveUntilSlot _ ->
-            0
-        RequireSomeOf _ ss ->
-            -- We don't know how many the user will sign with, so we just assume
-            -- the worst case of "signs with all".
-            sumVia scriptRequiredKeySigs ss
+        = fromIntegral $
+        maybe 0 estimateMinWitnessRequiredPerInput txPaymentTemplate
```

Comment:
No, for estimateTxSize we want to use estimate**Max**WitnessRequiredPerInput, right?